lib/intl.mjs

```diff
@@ -41,21 +43,44 @@ const descriptor = (value) => {
 const IntlDateTimeFormat = globalThis.Intl.DateTimeFormat;
 const ObjectAssign = Object.assign;
 
-export function DateTimeFormat(locale = IntlDateTimeFormat().resolvedOptions().locale, options = {}) {
+// Construction of  built-in Intl.DateTimeFormat objects is sloooooow,
+// so we'll only create those instances when we need them.
+// See https://bugs.chromium.org/p/v8/issues/detail?id=6528
+function getPropLazy(obj, prop) {
+  let val = obj[prop];
+  if (prop === TZ_RESOLVED) {
+    // for TimeZone, the initializer is a string, not a function
+    if (typeof val === 'string') {
+      val = new TimeZone(val);
+      obj[prop] = val;
+    }
+  } else if (typeof val === 'function') {
+    val = new IntlDateTimeFormat(obj[LOCALE], val(obj[OPTIONS]));
+    obj[prop] = val;
+  }
+  return val;
+}
+
+export function DateTimeFormat(locale, options) {
```

Comment:
```suggestion
export function DateTimeFormat(locale = undefined, options = {}) {
```

Reply:
Why is this change better? I'm not opposed, just wondering.

Reply:
It affects `DurationFormat.length`, which should be zero because the arguments are marked as optional in the spec

Reply:
s/`DurationFormat`/`DateTimeFormat`

Reply:
Ahhh, makes sense. What's funny is that I was waiting to send the equivalent PR to proposal-temporal until this one was approved... but the Test262 tests would have caught the problem over there. Any got ideas (see #4) for easily running Test262 against this repo? Each time I think I understand ECMAScript, I log onto this website and am reminded that I am still a padawan. 

My latest commit fixes the problem you found above, but while fixing it I found another problem: I wasn't fully protecting against the caller mutating `options` because I was only shallow-cloning it, so if someone did a wacky thing like {month: {toString: . Also I wasn't converting the `locale` param into a primitive to guard against later mutation. 

Working on a fix for that problem now.